neon_llm_fastchat/fastchat.py

```diff
@@ -54,19 +55,26 @@ def convert_role(role):
             role_fastchat = "Assistant"
         return role_fastchat
 
-    def ask(self, message, chat_history):
+    def assemble_prompt(self, message, chat_history):
         prompt = self.system_message
         # Context N messages
         for role, content in chat_history[-self.context_depth:]:
             role_fastchat = self.convert_role(role)
             prompt += f"### {role_fastchat}: {content}\n"
         prompt += f"### Human: {message}\n### Assistant:"
-        
+        return prompt
+
+    def ask(self, message, chat_history):
+        prompt = self.assemble_prompt(message, chat_history)
         bot_message = self.call_model(prompt)
         return bot_message
 
-    def call_model(self, prompt):
+    def tokenize(self, prompt):
```

Comment:
pls add type annotations including return types, as it is not clear what we expect

Reply:
Done with https://github.com/NeonGeckoCom/neon-llm-fastchat/pull/5/commits/b5901b5390a0f16ad9b54d940c1db952f6a71fae